File: plugin/src/shared/limits.ts
export const MAX_DEPTH = 6
export const MAX_INPUT_IDS = 2_000
export const MAX_PAGE_IDS = 100
export const MAX_IDENTIFIER_BYTES = 256
export const MAX_QUERY_BYTES = 1_024
export const MAX_DISPLAY_TEXT_BYTES = 1_024
export const MAX_VISITED_NODES = 10_000
export const MAX_RETURNED_NODES = 2_000
export const MAX_TEXT_BYTES = 8 * 1_024 * 1_024
export const MAX_ENVELOPE_BYTES = 24 * 1_024 * 1_024
export const MAX_RASTER_SIDE = 4_096
export const MAX_RASTER_PIXELS = 16_000_000
export const MAX_RASTER_DECODED_BYTES = 12 * 1_024 * 1_024
export const MAX_RASTER_BASE64_BYTES = 16 * 1_024 * 1_024
export const MAX_SVG_BYTES = 4 * 1_024 * 1_024
export const MAX_IN_FLIGHT = 4
export const MAX_QUEUE = 16
export const INACTIVITY_TIMEOUT_SECS = 15
export const TOTAL_TIMEOUT_SECS = 120
export const HEARTBEAT_SECS = 5
export const STALE_SESSION_SECS = 20
export const IDLE_GRACE_SECS = 30
export const CANCEL_CHECK_BATCH = 100
export const U32_MAX = 4_294_967_295

export const BROKER_URL = "ws://localhost:3056"
export const RECONNECT_DELAYS_MS: readonly [250, 500, 1_000, 2_000, 5_000] = [
  250, 500, 1_000, 2_000, 5_000,
]

File: crates/protocol/src/limits.rs
//! Reviewed, non-configurable upper limits for public protocol boundaries.

pub const MAX_DEPTH: u8 = 6;
pub const MAX_INPUT_IDS: usize = 2_000;
pub const MAX_PAGE_IDS: usize = 100;
pub const MAX_IDENTIFIER_BYTES: usize = 256;
pub const MAX_QUERY_BYTES: usize = 1_024;
pub const MAX_SEARCH_CURSOR_BYTES: usize = 64 * 1_024;
pub const MAX_DISPLAY_TEXT_BYTES: usize = 1_024;
pub const MAX_VISITED_NODES: usize = 10_000;
pub const MAX_RETURNED_NODES: usize = 2_000;
pub const MAX_TEXT_BYTES: usize = 8 * 1024 * 1024;
pub const MAX_ENVELOPE_BYTES: usize = 24 * 1024 * 1024;
pub const MAX_RASTER_SIDE: u32 = 4_096;
pub const MAX_RASTER_PIXELS: u64 = 16_000_000;
pub const MAX_RASTER_DECODED_BYTES: usize = 12 * 1024 * 1024;
pub const MAX_RASTER_BASE64_BYTES: usize = 16 * 1024 * 1024;
pub const MAX_SVG_BYTES: usize = 4 * 1024 * 1024;
pub const MAX_IN_FLIGHT: usize = 4;
pub const MAX_QUEUE: usize = 16;
pub const INACTIVITY_TIMEOUT_SECS: u64 = 15;
pub const TOTAL_TIMEOUT_SECS: u64 = 120;
pub const HEARTBEAT_SECS: u64 = 5;
pub const STALE_SESSION_SECS: u64 = 20;
pub const IDLE_GRACE_SECS: u64 = 30;
/// How long a call waits for the first election to install a backend before
/// giving up. The measured race is ~80µs — the MCP service starts answering
/// before `supervise` finishes electing — so this is orders of magnitude more
/// than the real case needs. It is capped so that a genuinely stuck election
/// still surfaces as an error rather than as latency: election retries from
/// 100ms to a 5s ceiling and can stay stuck for minutes.
pub const BACKEND_READY_MS: u64 = 1_000;
